pandas/io/excel.py

```diff
@@ -207,6 +210,10 @@ def __init__(self, io, **kwds):
         if engine is not None and engine != 'xlrd':
             raise ValueError("Unknown engine: %s" % engine)
 
+        # io can be a path object, just convert these to string (doesn't alter
+        # other objects)
+        io = _stringify_path(io)
```

Comment:
yeah, I actually really like to make this work with ``get_filepath_or_buffer`` directly, rather than special casing (and using this here). IIRC you put a comment why this wasn't going to work. But is there a way to modify things so that we can get it to work? 

Because this is the workhorse function and its a shame if anyhing has to deviate as things inevitably get missed.